### PR TITLE
Add verifier unit tests for ebpf samples

### PR DESCRIPTION
**Pull request by dthaler**
Due to issue #134, a bunch of samples that used to pass now fail, so to get this PR to pass I put them in the fail category.  To verify a fix for #134 the tests should be updated to put them back in the pass category.

The goal of this PR is to add tests to CI to catch such issues in the future.

Signed-off-by: Dave Thaler <dthaler@microsoft.com>

### CODE REVIEW

**Comment by elazarg** [comment]
Incredible.

Following #136, all but four tests should pass now. Can you amend accordingly?

**Comment by dthaler** [comment]
Updated tests and it passes again on my machine but I see a CI failure now.  Will investigate in an hour or two.

**Comment by dthaler** [comment]
CI is now fixed by this PR so ebpf-samples are available again and tests are passing.  This is ready to merge unless there's any more feedback.

File: src/test/test_verify.cpp [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+    TEST_SECTION("linux", "xdp2_kern.o", "xdp1");
+    TEST_SECTION("linux", "xdp2skb_meta_kern.o", "tc_mark");
+    TEST_SECTION("linux", "xdp2skb_meta_kern.o", "xdp_mark");
+    TEST_SECTION("linux", "xdpsock_kern.o", "xdp_sock");
+}
+
+// The following eBPF programs currently fail verification.
+// If the verifier is later updated to accept them, these should move
+// up to the previous test case.
+TEST_CASE("unverified linux samples", "[verify][linux][failure]") {
```
Comment (elazarg) [inline src/test/test_verify.cpp:293]: We should return an "unsupported feature" in these cases (map in map).
Just a note; I am going to merge this as it is.

**Comment by elazarg** [comment]
Thanks!

**Pull request merged**